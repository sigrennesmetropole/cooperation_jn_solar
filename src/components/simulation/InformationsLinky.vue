<script setup lang="ts">
import enedisSignature from '@/assets/illustrations/enedis-signature.svg'
import enedisSpace from '@/assets/illustrations/enedis-personal-space.svg'
import { apiEnedisService } from '@/services/api-enedis'
import { apiAutocalsolService } from '@/services/api-autocalsol'
import { getEnv, getEnedisSandboxPrm } from '@/services/env'
import { ref } from 'vue'

async function goToEnedisWebSite() {
  const url = await apiEnedisService.getUrlUserAuthorization()
  window.location.href = url
}

async function goToEnedisLogin() {
  if (getEnv() == 'dev') {
    await apiEnedisService.setPRMUser(getEnedisSandboxPrm())
    const consumption = await apiEnedisService.getAnnualConsumption()
    window.alert(JSON.stringify(consumption))
  } else if (getEnv() == 'prod') {
    const prm = await apiEnedisService.getPRMUser()
    if (prm === undefined) {
      await goToEnedisWebSite()
      return
    }
    const consumption = await apiEnedisService.getAnnualConsumption()
    window.alert(JSON.stringify(consumption))
  }
}

const isLoadingAutocalsol = ref(false)
async function callApiAutocalsol() {
  isLoadingAutocalsol.value = true
  const computeData = await apiAutocalsolService.getComputeData()
  isLoadingAutocalsol.value = false
  window.alert(JSON.stringify(computeData))
}
</script>

<template>
  <div
    class="flex flex-col gap-6 border border-slate-200 rounded-lg bg-slate-50 h-[451px] w-full px-4 py-6 font-dm-sans text-sm font-normal"
  >
    <p class="text-base font-medium">Comment connecter mon compteur Linky ?</p>
    <div class="flex flex-row gap-4">
      <img :src="enedisSignature" class="w-[110px] h-[34.43px] my-auto" />
      <p>Enedis gère le réseau d'électricité jusqu'au compteur.</p>
    </div>
    <p>
      Pour calculer vos économies d'énergie, nous devons accéder à vos données
      Linky gérées par Enedis.
    </p>
    <p>
      En cliquant sur ce bouton, vous accéderez à votre espace client Enedis où
      vous pourrez autoriser Enedis à nous transmettre votre consommation
      annuelle.
    </p>
    <img
      :src="enedisSpace"
      class="mx-auto cursor-pointer"
      @click="goToEnedisLogin()"
    />
    <p class="text-center">
      Vous pourrez changer d'avis et révoquer <br />votre autorisation à tout
      moment.
    </p>
  </div>

  <button @click="callApiAutocalsol()" class="border">
    <template v-if="isLoadingAutocalsol"> Loading </template>
    <template v-else> CALL API AUTOCALSOL </template>
  </button>
</template>
