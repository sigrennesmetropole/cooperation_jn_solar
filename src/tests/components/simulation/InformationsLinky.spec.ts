import InformationsLinky from '@/components/simulation/InformationsLinky.vue'
import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest'
import { mount, VueWrapper } from '@vue/test-utils'
import { createTestingPinia } from '@pinia/testing'
import nock from 'nock'

describe('InformationsLinky.vue', () => {
  let wrapper: VueWrapper

  beforeEach(async () => {
    wrapper = mount(InformationsLinky, {
      global: {
        plugins: [
          createTestingPinia({
            createSpy: vi.fn,
            stubActions: false,
            stubPatch: false,
            fakeApp: true,
          }),
        ],
      },
    })
  })

  afterEach(() => {
    jest.restoreAllMocks()
    nock.cleanAll()
  })

  it('renders the component correctly', () => {
    expect(wrapper.exists()).toBe(true)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('displays Enedis signature image', () => {
    const enedisSignatureImage = wrapper.find(
      'img[src$="enedis-signature.svg"]'
    )
    expect(enedisSignatureImage.exists()).toBe(true)
  })

  it('displays Enedis personal space image', () => {
    const enedisSpaceImage = wrapper.find(
      'img[src$="enedis-personal-space.svg"]'
    )
    expect(enedisSpaceImage.exists()).toBe(true)
  })
})
