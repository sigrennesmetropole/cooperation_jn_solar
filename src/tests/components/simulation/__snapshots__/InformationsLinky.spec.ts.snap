// Vitest Snapshot v1

exports[`InformationsLinky.vue > renders the component correctly 1`] = `
"<div class=\\"flex flex-col gap-6 border border-slate-200 rounded-lg bg-slate-50 h-[451px] w-full px-4 py-6 font-dm-sans text-sm font-normal\\">
  <p class=\\"text-base font-medium\\">Comment connecter mon compteur Linky ?</p>
  <div class=\\"flex flex-row gap-4\\"><img src=\\"/src/assets/illustrations/enedis-signature.svg\\" class=\\"w-[110px] h-[34.43px] my-auto\\">
    <p>Enedis gère le réseau d'électricité jusqu'au compteur.</p>
  </div>
  <p> Pour calculer vos économies d'énergie, nous devons accéder à vos données Linky gérées par Enedis. </p>
  <p> En cliquant sur ce bouton, vous accéderez à votre espace client Enedis où vous pourrez autoriser Enedis à nous transmettre votre consommation annuelle. </p><img tabindex=\\"0\\" src=\\"/src/assets/illustrations/enedis-personal-space.svg\\" class=\\"mx-auto cursor-pointer\\">
  <p class=\\"text-center\\"> Vous pourrez changer d'avis et révoquer <br>votre autorisation à tout moment. </p>
</div>"
`;
